internal/scheduleringester/schedulerdb.go

```diff
@@ -148,10 +149,36 @@ func (s *SchedulerDb) WriteDbOp(ctx context.Context, tx pgx.Tx, op DbOperation)
 			return errors.WithStack(err)
 		}
 	case MarkJobSetsCancelRequested:
-		jobSets := maps.Keys(o)
-		err := queries.MarkJobsCancelRequestedBySets(ctx, jobSets)
-		if err != nil {
-			return errors.WithStack(err)
+		for jobSetInfo, cancelDetails := range o {
+			if !cancelDetails.cancelQueued && !cancelDetails.cancelLeased {
+				continue
+			}
+			if cancelDetails.cancelQueued && cancelDetails.cancelLeased {
+				err := queries.MarkJobsCancelRequestedBySet(
+					ctx,
```

Comment:
I think this could be done in one query because it's effectively "update job set state = CANCELLED where queue = x and jobset = y and state in (x, y)"?

I wouldn't hold up this PR to change it but I'd definitely consider making the change as it'll be more efficient and (probably) less code.